x-pack/legacy/plugins/reporting/index.ts

```diff
@@ -70,6 +77,14 @@ export const reporting = (kibana: any) => {
       const pluginsSetup: ReportingSetupDeps = {
         usageCollection: server.newPlatform.setup.plugins.usageCollection,
       };
+
+      const fieldFormatServiceFactory = async (uiSettings: IUiSettingsClient) => {
+        const [, plugins] = await coreSetup.getStartServices();
```

Comment:
Hi, what does this do? This is not the only core service that we use. Why is it only necessary to implement this one using a callback?

Reply:
I talked to @alexwizp and we came to an understanding that this belongs in the existing ReportingPlugin#start method: https://github.com/elastic/kibana/blob/f0fbb99/x-pack/legacy/plugins/reporting/server/plugin.ts#L114

@alexwizp will make that change, so the Reporting team does not absorb the added tech debt and can keep focus on our other migration tasks.